streaming/consumer.cc

```diff
@@ -35,7 +36,9 @@ mutation_reader_consumer make_streaming_consumer(sstring origin,
 
             auto cf = db.local().find_column_family(reader.schema()).shared_from_this();
             auto guard = service::topology_guard(frozen_guard);
-            auto use_view_update_path = co_await db::view::check_needs_view_update_path(vb.local(), db.local().get_token_metadata_ptr(), *cf, reason);
+            bool use_view_update_path = co_await with_scheduling_group(db.local().get_gossip_scheduling_group(), [&] {
+                return db::view::check_needs_view_update_path(vb.local(), db.local().get_token_metadata_ptr(), *cf, reason);
+            });
```

Comment:
As mentioned above (or below, depends on github formatting), this throws data into a control-plane scheduling group, which can easily starve it. Or perhaps, cause it to starve other data-plane groups.

Switching groups can reduce throughput if there isn't enough concurrency, I don't know if that's the case here. Since it's in a consumer, probably it is.

Reply:
Alternatives can be:
 - add a new group (reasonable when we have [nested scheduling](https://github.com/scylladb/seastar/pull/2823), but doesn't solve the throughput problem if it exists)
 - teach the reader_concurrency_semaphore about query nesting

Reply:
Also, with nested queries, you can have the ABBA deadlock, across nodes.

Maybe a "reserved" semaphore capacity can solve this: mainline consumers can take 9/10 of the semaphore's resources, but 1/10 is reserved for nested consumers. We still need to add a way to tag queries as "nested" ones.

Reply:
Yes, by nesting I meant reservations for the nested/inner queries.

It could also be done by having two semaphores, indexed by the level, with separate reservations.

And yes, we'd need to tag queries.

nested scheduling groups is attractive here because it tags requests, and because we already associate reader concurrency semaphores with scheduling groups, and because it's just pretty. But it doesn't backport and we shouldn't employ it just for that. @xemul